airflow/models/serialized_dag.py

```diff
@@ -251,7 +251,7 @@ def remove_deleted_dags(
                     cls.fileloc_hash.notin_(alive_fileloc_hashes),
                     cls.fileloc.notin_(alive_dag_filelocs),
                     or_(
-                        cls.processor_subdir is None,
+                        cls.processor_subdir.is_(None),
```

Comment:
While I know `is_` is the correct thing here, I think this gives us an opportunity to add an additional test to ensure we don't break it in the future

Reply:
Agreed, I added a dummy `processor_subdir` into the test to ensure we rely on `cls.processor_subdir.is_(None)` correctly filtering. Let me know if that is enough